lib/Target/Btor/BtorToBtorIRTranslation.cpp

```diff
@@ -0,0 +1,534 @@
+#include "Target/Btor/BtorToBtorIRTranslation.h"
+#include "Dialect/Btor/IR/BtorDialect.h"
+#include "Dialect/Btor/IR/BtorOps.h"
+
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinOps.h"
+#include "mlir/IR/Dialect.h"
+#include "mlir/Translation.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/MemoryBuffer.h"
+#include "llvm/Support/SourceMgr.h"
+
+#include <iostream>
+#include <string>
+
+using namespace mlir;
+using namespace mlir::btor;
+
+void Deserialize::parseModelLine(Btor2Line *l) {
+  reachedLines[l->id] = l;
+  switch (l->tag) {
+  case BTOR2_TAG_bad:
+    bads.push_back(l);
+    break;
+
+  case BTOR2_TAG_constraint:
+    constraints.push_back(l);
+    break;
+
+  case BTOR2_TAG_init:
+    inits[l->args[0]] = l;
+    break;
+
+  case BTOR2_TAG_input:
+    inputs.push_back(l);
+    break;
+
+  case BTOR2_TAG_next:
+    nexts[l->args[0]] = l;
+    break;
+
+  case BTOR2_TAG_state:
+    states.push_back(l);
+    break;
+
+  default:
+    break;
+  }
+}
+
+bool Deserialize::parseModelIsSuccessful() {
+  if (!modelFile)
+    return false;
+  model = btor2parser_new();
+  if (!btor2parser_read_lines(model, modelFile)) {
+    std::cerr << "parse error at: " << btor2parser_error(model) << "\n";
+    return false;
+  }
+  // register each line that has been parsed
+  auto numLines = btor2parser_max_id(model);
+  inits.resize(numLines, nullptr);
```

Comment:
this still seems like an overkill. The number of inits and states is expected to be much fewer than the total number of lines in btor file. 

I would imagine that `inits` and `states` would be indexed by state id (i.e., first latch, second latch, third latch, etc.) 

Maybe I need to look at an example btor file again...

Reply:
Removed it. Should be much simpler now.